src/main/java/com/shepherdjerred/capstone/common/lobby/PlayerSlots.java

```diff
@@ -67,4 +67,12 @@ PlayerSlots setPlayer(QuoridorPlayer playerId, Player player) {
     newQuoridorPlayerMap.put(playerId, player);
     return new PlayerSlots(newQuoridorPlayerMap, playerCount);
   }
+
+  PlayerSlots addPlayer(Player player) {
+    checkNotNull(player);
+
+    var newQuoridorPlayerMap = new HashMap<>(playerIdMap);
```

Comment:
Use the setPlayer method instead of mutating the map directly.

Reply:
I think I got it.